pkg/controller/apicast/apicast_logic_reconciler.go

```diff
@@ -140,7 +155,52 @@ func (r *APIcastLogicReconciler) getAdminPortalCredentialsSecret() (*v1.Secret,
 	return &adminPortalCredentialsSecret, err
 }
 
+func (r *APIcastLogicReconciler) reconcileGatewayConfigurationSecret() (v1.Secret, error) {
+	secret, err := r.gatewayConfigurationSecret()
+	if err != nil {
+		return v1.Secret{}, err
+	}
+
+	err = r.setOwnerReference(&secret)
+	if err != nil {
+		return v1.Secret{}, err
+	}
+	// TODO is it safe to update always or should we detect for difference
+	// in OwnerRef?
+	err = r.Client().Update(context.TODO(), &secret)
+	if err != nil {
+		return v1.Secret{}, err
+	}
+
+	return secret, nil
+}
+
+func (r *APIcastLogicReconciler) gatewayConfigurationSecret() (v1.Secret, error) {
+	gatewayConfigSecret := v1.Secret{}
+
+	if r.APIcastCR.Spec.GatewayConfigurationSecret.Name == "" {
+		return v1.Secret{}, fmt.Errorf("Name field not defined in secret %s", r.APIcastCR.Spec.GatewayConfigurationSecret)
+	}
+
+	gatewayConfigSecretNamespacedName := types.NamespacedName{
+		Name:      r.APIcastCR.Spec.GatewayConfigurationSecret.Name,
+		Namespace: r.APIcastCR.Namespace,
+	}
+	err := r.Client().Get(context.TODO(), gatewayConfigSecretNamespacedName, &gatewayConfigSecret)
+	if err != nil {
+		return v1.Secret{}, err
+	}
+	if _, ok := gatewayConfigSecret.Data["config.json"]; !ok {
+		return v1.Secret{}, fmt.Errorf("%s key not found in secret %s", "config.json", gatewayConfigSecret.Name)
+	}
+	return gatewayConfigSecret, nil
+}
+
 func (r *APIcastLogicReconciler) reconcileAdminPortalCredentials() (*v1.Secret, bool, error) {
+	if r.APIcastCR.Spec.AdminPortalCredentialsRef == nil {
```

Comment:
If the secret has not been specified I return at this point.
We could make the if outside of this method and do it in the "Reconcile" method not sure which of the two options would be better. If we were to do it in reconcile then we would have to put an if there and perform the corresponding checks which I didn't like that much

Reply:
IMO it's ok calling this method and return early